### PR TITLE
sourcehut/update-4

### PR DESCRIPTION
**Pull request by eadwu**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Review by veprbl — COMMENTED** [review]
nix-review fails on NixOS:
```
builder for '/nix/store/p3kl7qfmzjc7s7plfr8pj5bh3cva2m9f-python3.7-srht-0.54.4.drv' failed with exit code 1; last 10 log lines:
  adding 'srht-0.54.4.dist-info/RECORD'
  removing build/bdist.linux-x86_64/wheel
  Finished executing setuptoolsBuildPhase
  installing
  Executing pipInstallPhase
  /build/core.sr.ht/dist /build/core.sr.ht
  Processing ./srht-0.54.4-py3-none-any.whl
  Requirement already satisfied: pgpy in /nix/store/plp8hjsrlrjs2ky2vz1qdj4sn0crxbh6-python3.7-pgpy-0.5.2/lib/python3.7/site-packages (from srht==0.54.4) (0.5.1)
  ERROR: Could not find a version that satisfies the requirement flask-login (from srht==0.54.4) (from versions: none)
  ERROR: No matching distribution found for flask-login (from srht==0.54.4)
```

File: pkgs/applications/version-management/sourcehut/todo.nix [hunk 1]
```diff
@@ -31,33 +31,13 @@ buildPythonPackage rec {
...
-      [meta.sr.ht]
-      origin=http://meta.sr.ht.local
-    '';
-  in ''
-    cp -f ${config} config.ini
-
-    # pytest tests fail
-    # pytest tests/
-  '';
+  dontUseSetuptoolsCheck = true;
```
[the remark this answers is not in the document]
Comment (eadwu) [reply pkgs/applications/version-management/sourcehut/todo.nix]: `doCheck` is not respected. 

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/hooks/setuptools-check-hook.sh#L15

**Comment by eadwu** [comment]
srht should build fine now, having problems with `python3.7-python-hglib-2.6.1` stalling on `installCheckPhase` now.

**Comment by veprbl** [comment]
@GrahamcOfBorg build sourcehut.buildsrht sourcehut.dispatchsrht sourcehut.gitsrht sourcehut.hgsrht sourcehut.listssrht sourcehut.mansrht sourcehut.metasrht sourcehut.pastesrht sourcehut.todosrht

**Pull request merged**